pkg/controllers/cloud/cloudprovideraccount_controller.go

```diff
@@ -64,19 +74,37 @@ func (r *CloudProviderAccountReconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{}, err
 	}
 
-	err = r.processCreate(&req.NamespacedName, providerAccount)
-	if err != nil {
+	if err = r.processCreate(&req.NamespacedName, providerAccount); err != nil {
 		_ = r.processDelete(&req.NamespacedName)
+		return ctrl.Result{}, err
 	}
 
-	return ctrl.Result{}, err
+	updatePendingSync(&r.pendingSyncCount, CPAController)
+	return ctrl.Result{}, nil
 }
 
 func (r *CloudProviderAccountReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	r.accountProviderType = make(map[types.NamespacedName]common.ProviderType)
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&cloudv1alpha1.CloudProviderAccount{}).
-		Complete(r)
+	if err := ctrl.NewControllerManagedBy(mgr).For(&cloudv1alpha1.CloudProviderAccount{}).Complete(r); err != nil {
+		return err
+	}
+	return mgr.Add(r)
+}
+
+// Start performs the initialization of the controller.
+// A controller is said to be initialized only when the dependent controllers
+// are synced, and it keeps a count of pending CRs to be reconciled.
+func (r *CloudProviderAccountReconciler) Start(stop context.Context) error {
+	r.Log.Info("Waiting for shared informer caches to be synced")
+	if !(*r.Mgr).GetCache().WaitForCacheSync(context.TODO()) {
+		return fmt.Errorf("failed to sync shared informer cache")
+	}
+
+	if err := setPendingCount(r.Client, &cloudv1alpha1.CloudProviderAccountList{}, &r.pendingSyncCount, CPAController); err != nil {
```

Comment:
nit: setCRPendingSyncCount... And based on return value, set controller status..